### PR TITLE
feat(bridge): enhance activity derived state for bridge orders

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

  **Fix bridge order status inconsistency showing "Open" instead of "Filled" when bridge completes**

https://github.com/user-attachments/assets/934617b7-593c-486b-b179-bba7b9ec6c84

  Bridge orders (swap + bridge) were displaying incorrect status in the main activity list. While individual progress sections showed correct
  completion status, the main status label incorrectly displayed "Open" instead of "Filled" for completed bridge orders.

  **Root cause**: Basic activity state calculation only considered swap status, ignoring bridge completion status.

  **Solution**: Created dedicated enhancement hook that combines swap and bridge status for accurate bridge order status display. Also improved
  loading state UX to use grey color with shimmer effect instead of confusing orange "pending" appearance.

  ## Status Mapping Logic

  The enhancement hook now maps combined swap + bridge states to accurate activity status:

  | Swap Status | Bridge Status | Final Activity Status | Display Label | UI Appearance |
  |-------------|---------------|----------------------|---------------|---------------|
  | ✅ Filled | 🔄 Loading/Fetching | `LOADING` | "Loading..." | Grey + shimmer |
  | ✅ Filled | 🔄 `PENDING` | `PENDING` | "Open" | Grey |
  | ✅ Filled | 🔄 `DEFAULT` | `PENDING` | "Open" | Grey |
  | ✅ Filled | ✅ `DONE` | `CONFIRMED` | "Filled" ✅ | Green |
  | ✅ Filled | ❌ `FAILED` | `FAILED` | "Failed" | Red |
  | ✅ Filled | ❌ `REFUND_COMPLETE` | `FAILED` | "Failed" | Red |
  | ❌ Failed | N/A | `FAILED` | "Failed" | Red |
  | 🔄 Pending | N/A | `PENDING` | "Open" | Grey |

 
  # To Test

  1. **Create a bridge order** (swap + bridge to different chain)
     - [ ] Verify order appears in activity list initially as "Pending" (grey)
     - [ ] Wait for swap to complete on source chain
     - [ ] **New**: Verify status shows "Loading..." with **grey color and shimmer effect** during bridge data fetch
     - [ ] Wait for bridge to complete on destination chain
     - [ ] **Main test**: Verify status shows "Filled" (green, not "Open")

  2. **Verify loading state UX**
     - [ ] Loading state shows grey color (same as "Open" status)
     - [ ] Loading state displays shimmer animation
     - [ ] Loading state shows "Loading..." text
     - [ ] No orange "pending" color during loading

  3. **Verify bridge progress sections**
     - [ ] Swap section shows checkmark when complete
     - [ ] Bridge section shows checkmark when complete
     - [ ] "You received" shows final amount on destination chain
     - [ ] Bridge transaction link is available and working

  4. **Verify non-bridge orders unaffected**
     - [ ] Regular swap orders still show correct status
     - [ ] Transaction activities still show correct status
     - [ ] No regression in activity list performance

  5. **Test edge cases**
     - [ ] Bridge order with failed bridge shows "Failed"
     - [ ] Bridge order with refund shows "Failed"
     - [ ] Bridge order with swap failure shows "Failed"
     - [ ] Loading state appears during bridge data fetch with proper styling

  # Background

  The issue occurred because `useActivityDerivedState` only calculated basic swap status, while bridge completion status was available
  separately in `useSwapAndBridgeContext`. The fix introduces `useEnhancedActivityDerivedState` that:

  1. **Internally fetches bridge data** for bridge orders using `useSwapAndBridgeContext`
  2. **Combines swap + bridge status** into unified completion state
  3. **Handles loading states** during bridge data fetching with improved UX
  4. **Preserves existing behavior** for non-bridge orders

  **Status calculation flow**:
  1. Non-bridge orders → Return original `activityDerivedState` (unchanged)
  2. Bridge orders + loading → Return `LOADING` status with loading flags
  3. Bridge orders + no context → Return original state (fallback)
  4. Bridge orders + bridge context → Override `isConfirmed`/`isFailed`/`isPending` based on `SwapAndBridgeStatus`

  **Key files**:
  - `useEnhancedActivityDerivedState.ts` - New hook for bridge-aware status calculation
  - `useSwapAndBridgeContext.ts` - Enhanced to return loading state
  - `ActivityDetails.tsx` - Simplified to use new enhancement hook
  - `StatusDetails.tsx` - Added `isLoading` prop for shimmer effect
  - Activity status enums - Added `LOADING` state for better UX
  - `index.tsx` - Updated color mapping for loading state
  - `styled.ts` - Enhanced shimmer effect logic and color handling

  **Architecture**: Uses enhancement hook pattern (not wrapper) that adds meaningful bridge-specific logic while maintaining clean separation of
   concerns.

## Summary by CodeRabbit

* **New Features**
  * Enhanced activity status now includes a loading state for bridge orders, improving status accuracy for cross-chain transactions.
  * Added conditional loading UI improvements for bridge activity summaries.
  * Introduced a progress link component with enhanced interaction feedback.

* **Bug Fixes**
  * Status labels and color indicators now correctly represent the new loading state.
  * Token logo rendering for bridge orders prevents UI flashing by showing cached or confirmed tokens appropriately.

* **Improvements**
  * Unified and up-to-date activity details and status for both swap and bridge orders.
  * Bridge order status updates triggered on every status change for real-time accuracy.
  * Visual status indicators now include shimmer effects during loading states.
  * Progress bar visibility logic refined to better reflect combined swap and bridge order states.
  * Status details component refactored for clarity and consistent activity state display.
  * Enhanced styling and interaction feedback for status labels and progress links.
  * Added loading indicators and refined status icons for clearer transaction state communication.
  * Improved fallback token data handling to maintain bridge functionality during data loading.
  * Removed deprecated bridge order loading component to streamline UI.
  * Destination chain name in bridge summaries now only shows when confirmed bridge data is available.

### CODE REVIEW

**Comment by elena-zh** [comment]
> Could it be related to the same issue here #5987 ? I see the token pair has mainnet on both ends. Cc @shoom3301

Yes, the issue with chains is the same as in #5987. 
However, it should not be related to the order status: it should be 'loading' instead of 'open'

**Comment by shoom3301** [comment]
@fairlighteth , I believe it happens because:
 - since localStorage is clean, PendingOrderUpdater doesn't update any order (pending bridge orders are only stored in localStorage)

Anyway, it's just a hypotesis, need to debug

**Review by elena-zh — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx [hunk 1]
```diff
@@ -59,92 +147,66 @@ export type StatusDetailsProps = {
...
 
-      {showCancellationModal && isCancellable && !isCancelled && (
+      {shouldShowStatusLabelBelow && (
         <StatusLabelBelow>
-          <CancelButton onClick={showCancellationModal} />
+          {showCancelButton && <CancelButton onClick={showCancellationModal} />}
+          {showProgressBar && !isLoading && (
+            <ProgressLink
+              role="button"
+              onClick={(e) => {
```
Comment (cowdan) [inline apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx]: nitpick: if the callback is more than a line, I always prefer to extract it for readability

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx [hunk 2]
```diff
@@ -10,45 +12,131 @@ import { Command } from '@cowprotocol/types'
...
+  if (isConfirmed) {
+    return <SVG src={OrderCheckImage} description="Order Filled" />
+  }
+  if (isExpired && isTransaction) {
+    return <SVG src={OrderCancelledImage} description="Transaction Failed" />
+  }
+  if (isFailed) {
+    return <SVG src={OrderCancelledImage} description="Failed" />
+  }
+  if (isExpired) {
```
Comment (cowdan) [inline apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx:78]: for readability, should we move `if (isExpired)` just after `if (isExpired && isTransaction)`?

**Review by cowdan — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx (hunk 2 above)
Comment (fairlighteth) [reply apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx:78]: Thank you, addressed now.

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx (hunk 1 above)
Comment (fairlighteth) [reply apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx]: Thank you, addressed now.

**Comment by fairlighteth** [comment]
@elena-zh pushed some changes, could you test again (when the build is ready).

**Comment by fairlighteth** [comment]
Requesting another review from you @shoom3301 given I had to do some code changes because of having merge conflicts, twice.

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/styled.ts [hunk 3]
```diff
@@ -237,7 +239,9 @@ export const StatusLabel = styled.div<{
         ? `var(${UI.COLOR_SUCCESS})`
         : color === 'danger'
           ? `var(${UI.COLOR_DANGER})`
-          : `var(${UI.COLOR_ALERT})`};
+          : color === ActivityState.OPEN
```
Comment (cowdan) [inline apps/cowswap-frontend/src/modules/account/containers/Transaction/styled.ts:242]: we should find a more scalable and easier way to combine logic and presentation...

**Review by cowdan — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx [hunk 4]
```diff
@@ -378,11 +388,38 @@ export function ActivityDetails(props: {
...
+                  }
+
+                  // Show spinner when waiting for API data in fresh sessions (don't use CurrencyLogoPair)
+                  return (
+                    <>
+                      <TokenLogo token={inputToken} size={32} />
+                      <IconSpinner spinnerWidth={1} margin="0 0 0 -4px" size={30} />
+                    </>
+                  )
+                })()
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx:419]: Is it possible to do this with a new component instead of an inline fn?
This could cause re-render issues if I'm not mistaken

cc @shoom3301 to confirm

**Review by alfetopito — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx (hunk 4 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx:419]: 100%, it must be a component instead of IIFE
@fairlighteth
